---
title: "1-preprocessing"
author: 
- name: Pierre-Luc Germain
  affiliation:
  - &IMLS Institute for Molecular Life Sciences, University of Zurich, Switzerland
- name: Helena L. Crowell
  affiliation: 
  - *IMLS
  - Swiss Institute of Bioinformatics (SIB), University of Zurich, Switzerland
date: "2019-10-23"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

# Introduction

This file combines files created by Pierre-Luc Germain and Helena L. Crowell:

* [preprocessing](https://github.com/HelenaLC/muscat-comparison/blob/master/MAGL/analysis/0-preprocessing.Rmd) 

* [clustering](https://github.com/HelenaLC/muscat-comparison/blob/master/MAGL/analysis/1-clustering.Rmd) 

* [annotation](https://github.com/HelenaLC/muscat-comparison/blob/master/MAGL/analysis/2-annotation.Rmd)

# Load packages

```{r}
suppressPackageStartupMessages({
  library(cowplot)
  library(Seurat)
  library(scater)
  library(scds)
  library(DropletUtils)
  library(readxl)
  library(SingleCellExperiment)
  library(muscat)
  library(Matrix)
  library(dplyr)
  library(ComplexHeatmap)
  library(viridis)
  library(ggplot2)
})

```

```{r}
# increase future's maximum allowed size of objects
# to be exported from default of 500 MB to 2 GB
options(future.globals.maxSize = 2048 * 1024 ^ 2)
```


# Doublet removal
```{r}
# load raw counts
fastq_dirs <- list.dirs("data", recursive = FALSE, full.names = TRUE)
names(fastq_dirs) <- basename(fastq_dirs)
sce <- DropletUtils::read10xCounts(fastq_dirs)

# rename row/colData colnames & SCE dimnames
names(rowData(sce)) <- c("ENSEMBL", "SYMBOL")
names(colData(sce)) <- c("sample_id", "barcode")
sce$sample_id <- factor(basename(sce$sample_id))
dimnames(sce) <- list(
    with(rowData(sce), paste(ENSEMBL, SYMBOL, sep = ".")),
    with(colData(sce), paste(barcode, sample_id, sep = ".")))

# load metadata
md_dir <- file.path("data", "metadata.xlsx")
md <- readxl::read_excel(md_dir)
m <- match(sce$sample_id, md$`Sample ID`)
sce$group_id <- md$Characteristics[m]

# remove undetected genes
sce <- sce[Matrix::rowSums(counts(sce) > 0) > 0, ]
dim(sce)
```

# Calculate QC Metrics
```{r}
(mito <- grep("mt-", rownames(sce), value = TRUE))
sce <- calculateQCMetrics(sce, feature_controls = list(Mt = mito))
# plotHighestExprs(sce, n = 20)
```

# Filtering

```{r}
# get sample-specific outliers
cols <- c("total_counts", "total_features_by_counts", "pct_counts_Mt")
log <- c(TRUE, TRUE, FALSE)
type <- c("both", "both", "higher")

drop_cols <- paste0(cols, "_drop")
for (i in seq_along(cols))
    colData(sce)[[drop_cols[i]]] <- isOutlier(sce[[cols[i]]], 
        nmads = 2.5, type = type[i], log = log[i], batch = sce$sample_id)

sapply(drop_cols, function(i) 
    sapply(drop_cols, function(j)
        sum(sce[[i]] & sce[[j]])))

cd <- data.frame(colData(sce))
ps <- lapply(seq_along(cols), function (i) {
    p <- ggplot(cd, aes_string(x = cols[i], alpha = drop_cols[i])) +
        geom_histogram(bins = 100, show.legend = FALSE) +
        scale_alpha_manual(values = c("FALSE" = 1, "TRUE" = 0.4)) +
        facet_wrap(~sample_id, ncol = 1, scales = "free") + 
        theme_classic() + theme(strip.background = element_blank())
    if (log[i]) 
        p <- p + scale_x_log10()
    return(p)
})
plot_grid(plotlist = ps, ncol = 3)


layout(matrix(1:2, nrow = 1))
ol <- Matrix::rowSums(as.matrix(colData(sce)[drop_cols])) != 0
x <- sce$total_counts
y <- sce$total_features_by_counts
LSD::heatscatter(x, y, log="xy", main = "unfiltered", 
    xlab = "Total counts", ylab = "Non-zero features")
LSD::heatscatter(x[!ol], y[!ol], log="xy", main = "filtered", 
    xlab = "Total counts", ylab = "Non-zero features")


# summary of cells kept
ns <- table(sce$sample_id)
ns_fil <- table(sce$sample_id[!ol])
print(rbind(
    unfiltered = ns, filtered = ns_fil, 
    "%" = ns_fil / ns * 100), digits = 0)


# drop outlier cells
sce <- sce[, !ol]
dim(sce)


# require count > 1 in at least 20 cells
sce <- sce[Matrix::rowSums(counts(sce) > 1) >= 20, ]
dim(sce)

saveRDS(sce, file.path("output", "MAGL-SCE.rds"))
```

# Integration
```{r warning=FALSE}
# create SeuratObject
so <- CreateSeuratObject(
    counts = counts(sce),
    meta.data = data.frame(colData(sce)),
    project = "10xMAGL")

# split by sample
cells_by_sample <- split(colnames(sce), sce$sample_id)
so <- lapply(cells_by_sample, function(i)
    SubsetData(so, cells = i))

# normalize, find variable genes, and scale
so <- lapply(so, NormalizeData, verbose = FALSE)
so <- lapply(so, FindVariableFeatures, nfeatures = 2e3,
    selection.method = "vst", do.plot = FALSE, verbose = FALSE)
so <- lapply(so, ScaleData, verbose = FALSE)

# find anchors & integrate
as <- FindIntegrationAnchors(so, verbose = FALSE)
```

```{r eval=FALSE, echo=FALSE}
#saveRDS(as, file = "output/as.rds")
```


```{r}
so <- IntegrateData(anchorset = as, dims = seq_len(30), verbose = FALSE)

# scale integrated data
DefaultAssay(so) <- "integrated"
so <- ScaleData(so, display.progress = FALSE)
```

# Dimension reduction
```{r}
so <- RunPCA(so, npcs = 30, verbose = FALSE)
so <- RunTSNE(so, reduction = "pca", dims = seq_len(20),
    seed.use = 1, do.fast = TRUE, verbose = FALSE)
so <- RunUMAP(so, reduction = "pca", dims = seq_len(20),
    seed.use = 1, verbose = FALSE)
```

# Clustering
```{r}
so <- FindNeighbors(so, reduction = "pca", dims = seq_len(20), verbose = FALSE)
for (res in c(0.1, 0.2, 0.4, 0.8, 1, 1.2, 2))
    so <- FindClusters(so, resolution = res, random.seed = 1, verbose = FALSE)
```

## DR colored by sample, group, and cluster ID
```{r}

thm <- theme(aspect.ratio = 1, legend.position = "none")
ps <- lapply(c("sample_id", "group_id", "ident"), function(u) {
    p1 <- DimPlot(so, reduction = "tsne", group.by = u) + thm
    p2 <- DimPlot(so, reduction = "umap", group.by = u)
    lgd <- get_legend(p2)
    p2 <- p2 + thm
    list(p1, p2, lgd)
    plot_grid(p1, p2, lgd, nrow = 1,
        rel_widths = c(1, 1, 0.5))
})
plot_grid(plotlist = ps, ncol = 1)
```


## Heatmap on known markers
```{r}
# set cluster IDs to resolution 0.1 clustering
so <- SetIdent(so, value = "integrated_snn_res.0.1")
unique(Idents(so))
```

```{r}
DefaultAssay(so) <- "RNA"
so <- SetIdent(so, value = "integrated_snn_res.0.1")
sce <- as.SingleCellExperiment(so, assay = "RNA")
colData(sce) <- as.data.frame(colData(sce)) %>% 
    mutate_if(is.character, as.factor) %>% 
    DataFrame(row.names = colnames(sce))
sce$cluster_id <- Idents(so)

known_markers <- list(
    astrocytes = c("Aqp4", "Gfap", "Fgfr3"),
    endothelial = c("Cldn5","Nostrin"),
    microglia = c("C1qb","Tyrobp"),
    neuronal = c("Snap25", "Stmn2"),
    neuronal_excitatory = "Slc17a7",
    neuronal_inhibitory = "Gad1",
    oligodendrocyte = "Opalin",
    OPC = "Pdgfra")

known_markers <- lapply(known_markers, sapply, function(g) 
    grep(paste0(g, "$"), rownames(sce), value = TRUE))

gs <- gsub(".*\\.", "", unlist(known_markers))
ks <- rep.int(names(known_markers), vapply(known_markers, length, numeric(1)))
labs <- sprintf("%s(%s)", gs, ks)

# split cells by cluster
cs_by_k <- split(colnames(sce), sce$cluster_id)

# compute cluster-marker means
means_by_cluster <- lapply(known_markers, function(gs)
    vapply(cs_by_k, function(i)
        Matrix::rowMeans(logcounts(sce)[gs, i, drop = FALSE]), 
        numeric(length(gs))))

# prep. for plotting & scale b/w 0 and 1
mat <- do.call("rbind", means_by_cluster)
mat <- muscat:::.scale(mat)    

rownames(mat) <- gs

cols <- muscat:::.cluster_colors[seq_along(known_markers)]
cols <- setNames(cols, names(known_markers))
row_anno <- rowAnnotation(
    df = data.frame(label = ks),
    col = list(label = cols),
    gp = gpar(col = "white")) 
Heatmap(mat,
        name = "scaled avg.\nexpression",
        col = viridis(10),
        cluster_rows = FALSE,
        cluster_columns = FALSE,
        row_names_side = "left",
        column_title = "cluster_id",
        column_title_side = "bottom",
        rect_gp = gpar(col = "white"),
        left_annotation = row_anno)
```


Below reproduces the heatmap generated in [here](https://htmlpreview.github.io/?https://raw.githubusercontent.com/HelenaLC/muscat-comparison/master/MAGL/docs/2-annotation.html#known_marker_genes). The cluster ID is different due to the version of seurat.

```{r}
Heatmap(mat,
        name = "scaled avg.\nexpression",
        col = viridis(10),
        cluster_rows = FALSE,
        column_order = c("0", "1", "2", "3", "5", 
                         "4", "7", "6", "9", "10",
                         "8", "12", "13", "11", "14",
                         "15", "17", "16", "18", "19"),
        row_names_side = "left",
        column_title = "cluster_id",
        column_title_side = "bottom",
        rect_gp = gpar(col = "white"),
        left_annotation = row_anno)

```


# Annotation
```{r}
# set cluster IDs to resolution 0.1 clustering
so <- SetIdent(so, value = "integrated_snn_res.0.1")

unique(Idents(so))

ao <- RenameIdents(so, 
                   `3` = "Astrocytes",
                   `11` = "Endothelial",
                   `16` = "Microglia", 
                   `5` = "Oligodendrocytes",
                   `13` = "OPC",
                   `0` = "Excit. Neuron",
                   `1` = "Excit. Neuron",
                   `2` = "Excit. Neuron",
                   `4` = "Excit. Neuron",
                   `6` = "Excit. Neuron",
                   `7` = "Excit. Neuron",
                   `8` = "Excit. Neuron",
                   `14` = "Excit. Neuron",
                   `17` = "Excit. Neuron",
                   
                   `9` = "Inhib. Neuron",
                   `10` = "Inhib. Neuron",
                   `12` = "Inhib. Neuron",
                   `15` = "Inhib. Neuron",
                   
                   `18` = "CPE cells",
                   `19` = "CPE cells")
ao@meta.data$cell_type <- Idents(ao) 
```

# Save SeuratObject to .rds

```{r}
saveRDS(as, file = "output/as.rds")
saveRDS(ao, file.path("output", "ao.rds"))
```

