<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2020-04-20" />

<title>2-tree</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">LPS</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">2-tree</h1>
<h4 class="author">Ruizhu Huang</h4>
<address class="author_afil">
Institute for Molecular Life Sciences, University of Zurich, Switzerland<br><h4 class="date">2020-04-20</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-04-25
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>LPS/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.5.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200424code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200424)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200424code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200424)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomfionarhuangtreeclimbRarticletreedca13c42c4e181f3e5baf422645c82a3a9b2ef53targetblankdca13c4a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/fionarhuang/treeclimbR_article/tree/dca13c42c4e181f3e5baf422645c82a3a9b2ef53" target="_blank">dca13c4</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomfionarhuangtreeclimbRarticletreedca13c42c4e181f3e5baf422645c82a3a9b2ef53targetblankdca13c4a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  .gitignore
    Untracked:  LPS/
    Untracked:  treeclimbR_article.Rproj

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with <code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="load-packages" class="section level1">
<h1>Load packages</h1>
<pre class="r"><code>suppressPackageStartupMessages({
  library(Seurat)
  library(SingleCellExperiment)
  library(clustree)
  library(dplyr)
  library(ggtree)
  library(ggnewscale)
  library(cowplot)
  library(TreeSummarizedExperiment)
  library(TreeHeatmap)
})</code></pre>
</div>
<div id="load-data" class="section level1">
<h1>load data</h1>
<pre class="r"><code>ao &lt;- readRDS(file.path(&quot;output&quot;, &quot;ao.rds&quot;))
as &lt;- readRDS(file.path(&quot;output&quot;, &quot;as.rds&quot;))</code></pre>
<pre class="r"><code>ao_list &lt;- lapply(split(colnames(ao), ao$sample_id), FUN=function(x) subset(ao, cells=x))</code></pre>
</div>
<div id="type-markers" class="section level1">
<h1>Type markers</h1>
<div id="cannonical-markers" class="section level2">
<h2>Cannonical markers</h2>
<pre class="r"><code>known_markers &lt;- list(
    astrocytes = c(&quot;Aqp4&quot;, &quot;Gfap&quot;, &quot;Fgfr3&quot;),
    endothelial = c(&quot;Cldn5&quot;,&quot;Nostrin&quot;),
    microglia = c(&quot;C1qb&quot;,&quot;Tyrobp&quot;),
    neuronal = c(&quot;Snap25&quot;, &quot;Stmn2&quot;),
    neuronal_excitatory = &quot;Slc17a7&quot;,
    neuronal_inhibitory = &quot;Gad1&quot;,
    oligodendrocyte = &quot;Opalin&quot;,
    OPC = &quot;Pdgfra&quot;)
known_mk &lt;- match(unlist(known_markers), 
                  gsub(&quot;.*\\.&quot;, &quot;&quot;, rownames(ao@assays$RNA)))
known_mk &lt;- rownames(ao@assays$RNA)[known_mk]</code></pre>
</div>
<div id="computationally-identified-markers" class="section level2">
<h2>Computationally-identified markers</h2>
<p>For each cell type, we select 20 markers that have the highest absolute value of log fold change. This is performed separately for samples in control group (<code>WT</code>) to avoid selecting state-marker genes.</p>
<pre class="r"><code>sample_WT &lt;- ao@meta.data %&gt;% 
  filter(group_id == &quot;WT&quot;) %&gt;%
  select(sample_id) %&gt;%
  unlist() %&gt;%
  unique()

mk_list &lt;- lapply(ao_list[sample_WT], FUN = function(x){
  FindAllMarkers(x, logfc.threshold = 0.5,
                     verbose = FALSE)})
saveRDS(mk_list, file = &quot;temp/mk_list.rds&quot;)</code></pre>
<pre class="r"><code># Find top N markers
topN &lt;- 20

sel_list &lt;- lapply(mk_list, FUN = function(x) {
  x %&gt;%
    group_by(cluster) %&gt;%
    top_n(topN, abs(avg_logFC)) %&gt;%
    ungroup(cluster) %&gt;%
    select(gene) %&gt;%
    unlist() %&gt;%
    unique()})

# select markers in at least 2 samples (50%)
sel_u &lt;- unlist(sel_list)
length(unique(sel_u))</code></pre>
<pre><code>[1] 225</code></pre>
<pre class="r"><code>tb &lt;- table(sel_u)
sel &lt;- names(tb[tb &gt;= 2])
length(sel)</code></pre>
<pre><code>[1] 125</code></pre>
<pre class="r"><code># type markers: known markers &amp; selected markers
sum(known_mk %in% sel)</code></pre>
<pre><code>[1] 3</code></pre>
<pre class="r"><code>length(setdiff(known_mk, sel))</code></pre>
<pre><code>[1] 10</code></pre>
<pre class="r"><code>all_mk &lt;- unique(c(sel, known_mk))
length(all_mk)</code></pre>
<pre><code>[1] 135</code></pre>
<pre class="r"><code>saveRDS(all_mk, file = &quot;temp/all_mk.rds&quot;)

# integrate data (using type markers + default(anchor.feature))
ft &lt;- unique(c(as@anchor.features, all_mk))
ias &lt;- IntegrateData(anchorset = as, 
                    features.to.integrate = ft,
                    dims = 1:20, verbose = FALSE)

length(ias@assays$integrated@var.features)</code></pre>
<pre><code>[1] 2000</code></pre>
<pre class="r"><code>dim(ias@assays$integrated@data)</code></pre>
<pre><code>[1]  2011 26607</code></pre>
<pre class="r"><code>ias@assays$integrated@scale.data</code></pre>
<pre><code>     [,1]
[1,]   NA</code></pre>
<pre class="r"><code># scale data (using type markers)
ias &lt;- ScaleData(ias, features = ft, verbose = FALSE)
length(ias@assays$integrated@var.features)</code></pre>
<pre><code>[1] 2000</code></pre>
<pre class="r"><code>dim(ias@assays$integrated@scale.data)</code></pre>
<pre><code>[1]  2011 26607</code></pre>
<pre class="r"><code>dim(ias@assays$integrated@data)</code></pre>
<pre><code>[1]  2011 26607</code></pre>
<p>Cells are clustered using all type markers.</p>
<pre class="r"><code># run PCA using type markers
ias &lt;- RunPCA(ias, features = all_mk, 
             npcs = 30, verbose = FALSE)
saveRDS(ias, file = &quot;temp/ias.rds&quot;)

# t-SNE and Clustering
ias &lt;- RunUMAP(ias, reduction = &quot;pca&quot;, dims = 1:20, 
               verbose = FALSE)</code></pre>
<pre><code>Warning: The default method for RunUMAP has changed from calling Python UMAP via reticulate to the R-native UWOT using the cosine metric
To use Python UMAP via reticulate, set umap.method to &#39;umap-learn&#39; and metric to &#39;correlation&#39;
This message will be shown once per session</code></pre>
<pre class="r"><code>ias &lt;- FindNeighbors(ias, 
                    reduction = &quot;pca&quot;, dims = 1:20, 
                    verbose = FALSE)</code></pre>
</div>
</div>
<div id="tree-construction" class="section level1">
<h1>Tree construction</h1>
<p>Leaves of the tree are clusters generated specified resolution.</p>
<pre class="r"><code>reso &lt;- 6
ias &lt;- FindClusters(ias, resolution = reso, verbose = FALSE)
ias &lt;- BuildClusterTree(object = ias,
                        dims = 1:20)
cell_tree &lt;- Tool(object = ias, slot = &#39;BuildClusterTree&#39;)
cell_tree</code></pre>
<pre><code>
Phylogenetic tree with 66 tips and 65 internal nodes.

Tip labels:
    0, 1, 2, 3, 4, 5, ...

Rooted; includes branch lengths.</code></pre>
<pre class="r"><code>#saveRDS(cell_tree, file = &quot;output/cell_tree.rds&quot;)</code></pre>
</div>
<div id="viz-tree" class="section level1">
<h1>Viz tree</h1>
<p>Within cluster, cell counts in each cell type are calcuated.</p>
<pre class="r"><code># information: cell types + cluster ID
reso_vn &lt;- paste0(&quot;integrated_snn_res.&quot;, reso)
cell_info &lt;- ao@meta.data %&gt;% 
  select(&quot;sample_id&quot;, &quot;barcode&quot;, &quot;group_id&quot;, &quot;cell_type&quot;) %&gt;% 
  mutate(cell = rownames(ao@meta.data))  %&gt;% 
  left_join(ias@meta.data[, c(&quot;sample_id&quot;, &quot;barcode&quot;, &quot;group_id&quot;,
                      reso_vn)]) %&gt;%
  rename(cluster_id = !!reso_vn)</code></pre>
<pre><code>Joining, by = c(&quot;sample_id&quot;, &quot;barcode&quot;, &quot;group_id&quot;)</code></pre>
<pre class="r"><code># cell type of each leaf on the tree
aa &lt;- table(cell_info$cell_type, cell_info$cluster_id)
ind &lt;- apply(aa, 2, which.max)
cell_anno &lt;- data.frame(cluster_id = colnames(aa),
                        node = transNode(tree = cell_tree, 
                                         node = colnames(aa)),
                        cell_type = rownames(aa)[ind])
rownames(cell_anno) &lt;- cell_anno$cluster_id</code></pre>
<p>Leaves of the tree are annotated with cell types. The bar plot displays cell counts of cell types within each cluster. The cell type that has maximum counts in a cluster is used to annotate the leaf.</p>
<pre class="r"><code># Tree
Fig_0 &lt;- ggtree(cell_tree, ladderize = TRUE)

# Tree + cell type
d &lt;- dplyr::filter(Fig_0$data, isTip) %&gt;% select(c(label, y)) 

dd &lt;- cell_info %&gt;%
  rename(label = cluster_id) %&gt;%
  left_join(d, by=&#39;label&#39;)</code></pre>
<pre><code>Warning: Column `label` joining factor and character vector, coercing into
character vector</code></pre>
<pre class="r"><code>colr &lt;- c(&quot;Astrocytes&quot; = &quot;#E41A1C&quot;, &quot;CPE cells&quot; = &quot;#377EB8&quot;,
          &quot;Endothelial&quot; = &quot;#4DAF4A&quot;, &quot;Excit. Neuron&quot; = &quot;#984EA3&quot;,
          &quot;Inhib. Neuron&quot; = &quot;#FFFF33&quot;, &quot;Microglia&quot; = &quot;#FF7F00&quot;,
           &quot;Oligodendrocytes&quot; = &quot;#A65628&quot;, &quot;OPC&quot; = &quot;#F781BF&quot;)
bar &lt;- ggplot(dd, aes(x = y)) +
    geom_bar(aes(fill = cell_type)) +
    scale_fill_manual(values = colr) +
    theme(legend.position = &quot;none&quot;) +
    theme(axis.line.x=element_blank(),
          axis.text.y=element_blank(),
          axis.ticks.y=element_blank(),
          axis.title.x=element_blank(),
          axis.title.y=element_blank(),
          legend.position=&quot;none&quot;,
          panel.background=element_blank(),
          panel.border=element_blank(),
          panel.grid.major=element_blank(),
          panel.grid.minor=element_blank(),
          plot.background=element_blank())


bar_df &lt;- ggplot_build(bar)$data[[1]]
vf &lt;- bar_df$fill
names(vf) &lt;- vf

# flip x-y coordinates and scale x 
scale_bar &lt;- 1/300
ux &lt;- max(bar_df$ymax)
lx &lt;- min(bar_df$ymin)

# barplot: gap to the tree (2)
gap_tree &lt;- ceiling(max(Fig_0$data$x)) + 0.1

#
ct &lt;- names(colr)
names(ct) &lt;- colr
dat_bar &lt;- bar_df %&gt;%
  select(fill, x, xmin, xmax, ymin, ymax) %&gt;%
  rename(ymin = xmin,
         ymax = xmax, 
         xmin = ymin, 
         xmax = ymax,
         y = x) %&gt;%
  mutate(xmin_s = (xmin-lx)/(ux-lx)*ux*scale_bar+ gap_tree,
         xmax_s = (xmax-lx)/(ux-lx)*ux*scale_bar+ gap_tree,
         Cell_type = factor(ct[fill], levels = rev(ct)) )
  
   
# label cell counts
x_bar_label &lt;- c(0, 500, 1000, 1500)
names(x_bar_label) &lt;- as.character(x_bar_label*scale_bar + gap_tree)

dat_bar_ext &lt;- dat_bar %&gt;%
  mutate(xmin = min(xmin_s), xmax = max(xmax_s))


Fig_1 &lt;- Fig_0 + 
  geom_segment(data = dat_bar_ext, 
               aes(x = xmin, xend = xmax,
                   y = y, yend = y), 
               linetype = &quot;solid&quot;,
               color = &quot;grey80&quot;, size = 0.05, 
               show.legend = FALSE)+
  geom_rect(data = dat_bar,
            aes(xmin = xmin_s, ymin = ymin, 
                xmax = xmax_s, ymax = ymax, 
                fill = Cell_type), 
            inherit.aes = FALSE) +
  scale_fill_manual(values = colr) +
  geom_segment(data = dat_bar,
               aes (x = min(xmin_s), y = -0.5, 
                    xend = max(xmax_s), yend = -0.5),
               size = 0.4) +
  annotate(&quot;segment&quot;, 
           x = as.numeric(names(x_bar_label)), 
           y = rep(-0.2, length(x_bar_label)),
           xend = as.numeric(names(x_bar_label)), 
           yend = rep(-0.8, length(x_bar_label)),
           size = 0.5) +
  annotate(&quot;text&quot;, 
           x = as.numeric(names(x_bar_label)), 
           y = rep(-1.3, length(x_bar_label)),
           label= x_bar_label, size = 1.5) +
  annotate(&quot;text&quot;, x = mean(min(dat_bar$xmin_s),max(dat_bar$xmax_s)),
           y = max(dat_bar$ymax) + 3, label = &quot;Cell counts&quot;, 
           hjust =0, fontface = &quot;bold&quot;, size = 3) +
  guides(fill = guide_legend(order = 2, ncol = 2)) +
  new_scale_fill() +
  theme(legend.position = c(0.1, 0.6))


Fig_1 </code></pre>
<p><img src="figure/2-tree.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="output-se" class="section level1">
<h1>Output SE</h1>
<div id="type-genes" class="section level2">
<h2>Type genes</h2>
<p>The data of type genes is saved for later visualization as a <code>SingleCellExperiment</code> object.</p>
<pre class="r"><code># scale data of cell type marker
mkc &lt;- ias@assays$integrated@scale.data[all_mk, ]

# cell information
cell_dt &lt;- cbind.data.frame(
    cell = rownames(ias@meta.data),
    ias@meta.data[, c(&quot;sample_id&quot;, &quot;barcode&quot;, &quot;group_id&quot;,
                      reso_vn)]) %&gt;%
    rename(cluster_id = !!reso_vn) %&gt;% 
    left_join(cell_anno) </code></pre>
<pre><code>Joining, by = &quot;cluster_id&quot;</code></pre>
<pre class="r"><code>se_mk &lt;- SummarizedExperiment(assays = list(mkc), 
                               colData = cell_dt)
#saveRDS(se_mk, file = &quot;output/se_mk.rds&quot;)</code></pre>
</div>
<div id="genes-without-type-genes" class="section level2">
<h2>Genes without type genes</h2>
<p>The original data of genes that are not selected as type genes for all cells is output as a <code>SummarizedExperiment</code> object. The data is later aggregated the different levels of the tree to compare between groups <code>LPS</code> and <code>WT</code>.</p>
<pre class="r"><code>(test &lt;- all(rownames(ao@meta.data) == rownames(ias@meta.data)))</code></pre>
<pre><code>[1] TRUE</code></pre>
<pre class="r"><code>if (!test) {stop(&quot;the order of genes are changed&quot;)}

# the original data
DefaultAssay(ias) &lt;- &quot;RNA&quot;
count &lt;- ias@assays$RNA@counts
count &lt;- count[!rownames(count) %in% all_mk, ]

cell_info &lt;- ao@meta.data %&gt;% 
  select(&quot;sample_id&quot;, &quot;barcode&quot;, &quot;group_id&quot;, &quot;cell_type&quot;) %&gt;% 
  mutate(cell = rownames(ao@meta.data))  %&gt;% 
  left_join(ias@meta.data[, c(&quot;sample_id&quot;, &quot;barcode&quot;, &quot;group_id&quot;,
                      reso_vn)]) %&gt;%
  rename(cluster_id = !!reso_vn) </code></pre>
<pre><code>Joining, by = c(&quot;sample_id&quot;, &quot;barcode&quot;, &quot;group_id&quot;)</code></pre>
<pre class="r"><code>se_count &lt;- SummarizedExperiment(assays = list(count),
                                  colData = cell_info)</code></pre>
<pre class="r"><code>saveRDS(known_mk, file = &quot;output/known_mk.rds&quot;)
saveRDS(cell_tree, file = &quot;output/cell_tree.rds&quot;)
saveRDS(se_mk, file = &quot;output/se_mk.rds&quot;)
saveRDS(se_count, file = &quot;output/se_count.rds&quot;)
saveRDS(cell_info, file = &quot;output/cell_info.rds&quot;)
saveRDS(all_mk, file = &quot;output/all_mk.rds&quot;)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Mojave 10.14.4

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] TreeHeatmap_0.1.0              TreeSummarizedExperiment_1.3.0
 [3] cowplot_1.0.0                  ggnewscale_0.4.0              
 [5] ggtree_2.1.6                   dplyr_0.8.5                   
 [7] clustree_0.4.1                 ggraph_2.0.0                  
 [9] ggplot2_3.3.0                  SingleCellExperiment_1.8.0    
[11] SummarizedExperiment_1.16.0    DelayedArray_0.12.0           
[13] BiocParallel_1.20.0            matrixStats_0.55.0            
[15] Biobase_2.46.0                 GenomicRanges_1.38.0          
[17] GenomeInfoDb_1.22.0            IRanges_2.20.0                
[19] S4Vectors_0.24.0               BiocGenerics_0.32.0           
[21] Seurat_3.1.1                   workflowr_1.5.0               

loaded via a namespace (and not attached):
  [1] backports_1.1.6        plyr_1.8.5             igraph_1.2.4.1        
  [4] lazyeval_0.2.2         splines_3.6.1          listenv_0.7.0         
  [7] digest_0.6.25          htmltools_0.4.0        viridis_0.5.1         
 [10] gdata_2.18.0           fansi_0.4.1            magrittr_1.5          
 [13] cluster_2.1.0          ROCR_1.0-7             globals_0.12.4        
 [16] graphlayouts_0.5.0     RcppParallel_4.4.4     R.utils_2.9.0         
 [19] colorspace_1.4-1       ggrepel_0.8.1          xfun_0.11             
 [22] crayon_1.3.4           RCurl_1.95-4.12        jsonlite_1.6.1        
 [25] survival_2.44-1.1      zoo_1.8-6              ape_5.3               
 [28] glue_1.4.0             polyclip_1.10-0        gtable_0.3.0          
 [31] zlibbioc_1.32.0        XVector_0.26.0         leiden_0.3.1          
 [34] future.apply_1.3.0     scales_1.1.0           bibtex_0.4.2          
 [37] Rcpp_1.0.4             metap_1.1              viridisLite_0.3.0     
 [40] tidytree_0.3.3         reticulate_1.13        rsvd_1.0.2            
 [43] SDMTools_1.1-221.1     tsne_0.1-3             htmlwidgets_1.5.1     
 [46] httr_1.4.1             gplots_3.0.1.1         RColorBrewer_1.1-2    
 [49] ellipsis_0.3.0         ica_1.0-2              pkgconfig_2.0.3       
 [52] R.methodsS3_1.7.1      farver_2.0.3           uwot_0.1.4            
 [55] labeling_0.3           tidyselect_1.0.0       rlang_0.4.5           
 [58] reshape2_1.4.3         later_1.0.0            munsell_0.5.0         
 [61] tools_3.6.1            cli_2.0.2              ggridges_0.5.1        
 [64] evaluate_0.14          stringr_1.4.0          yaml_2.2.0            
 [67] npsurv_0.4-0           knitr_1.26             fs_1.3.1              
 [70] fitdistrplus_1.0-14    tidygraph_1.1.2        caTools_1.17.1.2      
 [73] purrr_0.3.3            RANN_2.6.1             pbapply_1.4-2         
 [76] future_1.15.0          nlme_3.1-142           R.oo_1.23.0           
 [79] aplot_0.0.4            compiler_3.6.1         plotly_4.9.1          
 [82] png_0.1-7              lsei_1.2-0             treeio_1.11.2         
 [85] tibble_3.0.0           tweenr_1.0.1           stringi_1.4.6         
 [88] RSpectra_0.15-0        lattice_0.20-38        Matrix_1.2-17         
 [91] vctrs_0.2.4            pillar_1.4.3           lifecycle_0.2.0       
 [94] BiocManager_1.30.10    Rdpack_0.11-0          lmtest_0.9-37         
 [97] RcppAnnoy_0.0.14       data.table_1.12.6      bitops_1.0-6          
[100] irlba_2.3.3            gbRd_0.4-11            httpuv_1.5.2          
[103] patchwork_1.0.0        R6_2.4.1               promises_1.1.0        
[106] KernSmooth_2.23-15     gridExtra_2.3          codetools_0.2-16      
[109] MASS_7.3-51.4          gtools_3.8.1           assertthat_0.2.1      
[112] rprojroot_1.3-2        withr_2.1.2            sctransform_0.2.0     
[115] GenomeInfoDbData_1.2.2 grid_3.6.1             tidyr_1.0.2           
[118] rmarkdown_1.17         rvcheck_0.1.8          Rtsne_0.15            
[121] git2r_0.26.1           ggforce_0.3.1         </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
