<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="fionarhuang" />

<meta name="date" content="2020-04-23" />

<title>Preparation</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">microbe</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="1-Preparation.html">Preparation</a>
</li>
<li>
  <a href="2-Analysis.html">Analysis</a>
</li>
<li>
  <a href="3-Visualization.html">Visualization</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://robinsonlabuzh.github.io">Robinson Lab</a>
</li>
<li>
  <a href="https://github.com/fionarhuang/treeclimbR_article">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Preparation</h1>
<h4 class="author">fionarhuang</h4>
<h4 class="date">2020-04-23</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-04-24
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>microbe/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.5.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200423code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200423)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200423code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200423)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomfionarhuangtreeclimbRarticletree672ffdb32770d8ede8594670881236ed8a1788catargetblank672ffdba"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/fionarhuang/treeclimbR_article/tree/672ffdb32770d8ede8594670881236ed8a1788ca" target="_blank">672ffdb</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomfionarhuangtreeclimbRarticletree672ffdb32770d8ede8594670881236ed8a1788catargetblank672ffdba" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  .gitignore
    Untracked:  miRNA/
    Untracked:  microbe/analysis/1-Preparation.Rmd
    Untracked:  microbe/analysis/2-Analysis.Rmd
    Untracked:  microbe/analysis/3-Visualization.Rmd
    Untracked:  treeclimbR_article.Rproj

Unstaged changes:
    Deleted:    microbe/analysis/Analysis.Rmd
    Deleted:    microbe/analysis/Preparation.Rmd
    Deleted:    microbe/analysis/Visualization.Rmd
    Modified:   microbe/analysis/_site.yml
    Modified:   microbe/analysis/index.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with <code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<pre class="r"><code>suppressPackageStartupMessages({
    library(curatedMetagenomicData)
    library(phyloseq)
    library(dplyr)
    library(ggplot2)
    library(ape)
})</code></pre>
<div id="download-data" class="section level1">
<h1>Download data</h1>
<pre class="r"><code>stool_0 &lt;- BackhedF_2015.metaphlan_bugs_list.stool()</code></pre>
<pre><code>snapshotDate(): 2019-10-22</code></pre>
<pre><code>see ?curatedMetagenomicData and browseVignettes(&#39;curatedMetagenomicData&#39;) for documentation</code></pre>
<pre><code>loading from cache</code></pre>
<pre class="r"><code>stool &lt;- ExpressionSet2phyloseq(stool_0, phylogenetictree = TRUE)</code></pre>
<pre><code>Warning: `data_frame()` is deprecated as of tibble 1.1.0.
Please use `tibble()` instead.
This warning is displayed once every 8 hours.
Call `lifecycle::last_warnings()` to see where this warning was generated.</code></pre>
</div>
<div id="data-preparation" class="section level1">
<h1>Data preparation</h1>
<p>Only samples from babies are used.</p>
<pre class="r"><code># sample variables: group (Infant (B1D, B4M, B12M, M)), 
# born_method(vaginal, c_section)
samp &lt;- data.frame(sample_data(stool), check.names = FALSE)
samp_data &lt;- samp %&gt;% 
    mutate(sample_id = rownames(samp),
           group_id = gsub(&quot;..*_&quot;, &quot;&quot;, sample_id),
         group_id = ifelse(group_id == &quot;B&quot;, &quot;Baby_1d&quot;,
                           ifelse(group_id == &quot;M&quot;, &quot;Mother&quot;,
                                  paste0(&quot;Baby_&quot;, tolower(group_id))
                           )),
         period = ifelse(grepl(&quot;_1d&quot;, group_id), &quot;0 M&quot;,
                         ifelse(grepl(&quot;_4m&quot;, group_id), &quot;4 M&quot;, 
                                      ifelse(grepl(&quot;_12m&quot;, group_id), 
                                             &quot;12 M&quot;, &quot;Mother&quot;)))) %&gt;%
    filter(family_role != &quot;mother&quot;) %&gt;%
  select(group_id, born_method, feeding_practice, born_method,
         period, number_reads, sample_id)

head(samp_data)  </code></pre>
<pre><code>  group_id born_method          feeding_practice period number_reads sample_id
1  Baby_1d     vaginal             mixed_feeding    0 M     33292394   SID10_B
2  Baby_4m     vaginal exclusively_breastfeeding    4 M     42937478  SID10_4M
3 Baby_12m     vaginal          no_breastfeeding   12 M     43463570 SID10_12M
4  Baby_1d     vaginal exclusively_breastfeeding    0 M     36337796   SID45_B
5  Baby_4m     vaginal exclusively_breastfeeding    4 M     43735250  SID45_4M
6 Baby_12m     vaginal         any_breastfeeding   12 M     34750848 SID45_12M</code></pre>
<pre class="r"><code>table(samp_data$group_id, samp_data$born_method)</code></pre>
<pre><code>          
           c_section vaginal
  Baby_12m        15      79
  Baby_1d         15      80
  Baby_4m         15      81</code></pre>
<p>The count table</p>
<pre class="r"><code># raw counts
otu_count &lt;- as.matrix(data.frame(otu_table(stool)))[, samp_data$sample_id]
reads &lt;- samp_data$number_reads
new_count &lt;- round((otu_count/100) %*% diag(reads), digits = 0)
colnames(new_count) &lt;- colnames(otu_count)</code></pre>
<p>Taxonomic table and phylogenetic tree are updated after the removement of some OTUs.</p>
<pre class="r"><code># taxonomic table
new_tax &lt;- data.frame(tax_table(stool)@.Data, check.names = FALSE)

# phylogenetic tree
new_phy &lt;- phy_tree(stool)</code></pre>
</div>
<div id="sequencing-depths" class="section level1">
<h1>Sequencing depths</h1>
<p>Sequencing depths are quite different among samples but quite similar between groups.</p>
<pre class="r"><code>ggplot(data = samp_data) +
  geom_density(aes(x = number_reads, y = ..density..,
                     fill = born_method, group = born_method),
               alpha = 0.4) +
    facet_grid(rows = vars(group_id)) +
  scale_x_continuous(breaks = scales::pretty_breaks(n = 10)) +
  scale_y_continuous(breaks = scales::pretty_breaks(n = 10))</code></pre>
<p><img src="figure/1-Preparation.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="save-outputs" class="section level1">
<h1>Save outputs</h1>
<p>Average samples are prepared for later visualization (e.g., heatmap) to avoid display so many samples in a figure. Samples, with the same born method and the same time point, are randomly assigned into 5 categories, and an average sample is generated for each category by taking the average counts of OTUs.</p>
<pre class="r"><code>save(new_count, otu_count, samp_data, new_phy, new_tax,
     file = &quot;output/Preparation.RData&quot;)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Mojave 10.14.4

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] parallel  stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] ape_5.3                       ggplot2_3.3.0                
 [3] phyloseq_1.30.0               curatedMetagenomicData_1.16.0
 [5] ExperimentHub_1.12.0          dplyr_0.8.5                  
 [7] Biobase_2.46.0                AnnotationHub_2.18.0         
 [9] BiocFileCache_1.10.2          dbplyr_1.4.2                 
[11] BiocGenerics_0.32.0           workflowr_1.5.0              

loaded via a namespace (and not attached):
 [1] nlme_3.1-142                  fs_1.3.1                     
 [3] bit64_0.9-7                   httr_1.4.1                   
 [5] rprojroot_1.3-2               tools_3.6.1                  
 [7] backports_1.1.6               R6_2.4.1                     
 [9] vegan_2.5-6                   mgcv_1.8-28                  
[11] DBI_1.0.0                     colorspace_1.4-1             
[13] permute_0.9-5                 ade4_1.7-13                  
[15] withr_2.1.2                   tidyselect_1.0.0             
[17] bit_1.1-14                    curl_4.2                     
[19] compiler_3.6.1                git2r_0.26.1                 
[21] cli_2.0.2                     scales_1.1.0                 
[23] rappdirs_0.3.1                stringr_1.4.0                
[25] digest_0.6.25                 rmarkdown_1.17               
[27] XVector_0.26.0                pkgconfig_2.0.3              
[29] htmltools_0.4.0               fastmap_1.0.1                
[31] rlang_0.4.5                   RSQLite_2.1.2                
[33] shiny_1.4.0                   farver_2.0.3                 
[35] jsonlite_1.6.1                magrittr_1.5                 
[37] biomformat_1.14.0             Matrix_1.2-17                
[39] Rcpp_1.0.4                    munsell_0.5.0                
[41] S4Vectors_0.24.0              Rhdf5lib_1.8.0               
[43] fansi_0.4.1                   lifecycle_0.2.0              
[45] stringi_1.4.6                 yaml_2.2.0                   
[47] MASS_7.3-51.4                 zlibbioc_1.32.0              
[49] rhdf5_2.30.0                  plyr_1.8.5                   
[51] grid_3.6.1                    blob_1.2.0                   
[53] promises_1.1.0                crayon_1.3.4                 
[55] lattice_0.20-38               Biostrings_2.54.0            
[57] splines_3.6.1                 multtest_2.42.0              
[59] knitr_1.26                    pillar_1.4.3                 
[61] igraph_1.2.4.1                reshape2_1.4.3               
[63] codetools_0.2-16              stats4_3.6.1                 
[65] glue_1.4.0                    BiocVersion_3.10.1           
[67] evaluate_0.14                 data.table_1.12.6            
[69] BiocManager_1.30.10           vctrs_0.2.4                  
[71] httpuv_1.5.2                  foreach_1.4.7                
[73] gtable_0.3.0                  purrr_0.3.3                  
[75] tidyr_1.0.2                   assertthat_0.2.1             
[77] xfun_0.11                     mime_0.7                     
[79] xtable_1.8-4                  later_1.0.0                  
[81] survival_2.44-1.1             tibble_3.0.0                 
[83] iterators_1.0.12              AnnotationDbi_1.48.0         
[85] memoise_1.1.0                 IRanges_2.20.0               
[87] cluster_2.1.0                 ellipsis_0.3.0               
[89] interactiveDisplayBase_1.24.0</code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
